### PR TITLE
fix: TOC links

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview
Tweak to how Gamut generates page links + fixes previous broken Table of Contents links like Button + Theme

### PR Checklist

- [x] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/styleguide/.storybook/components/TableOfContents/utils.tsx [hunk 1]
```diff
@@ -57,6 +57,9 @@ export function useKind(kind: string = '') {
   const { storyStore } = useContext(DocsContext);
   const kindStore: Record<string, Kind> = storyStore?._kinds ?? {};
   const kindMeta = storyStore?._kinds?.[kind];
+  const storiesStore = Object.keys(storyStore?._stories);
+  console.log('Store', storiesStore);
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/.storybook/components/TableOfContents/utils.tsx:61]:
```suggestion
```

Comment (codecaaron) [inline packages/styleguide/.storybook/components/TableOfContents/utils.tsx:61]:
```suggestion
  const allStories = useMemo(() => Object.keys(storyStore?._stories), [storyStore?._stories])
```

**Review by codecaaron — CHANGES_REQUESTED** [review]

File: packages/styleguide/.storybook/components/TableOfContents/utils.tsx (hunk 1 above)
Comment (dreamwasp) [reply packages/styleguide/.storybook/components/TableOfContents/utils.tsx]: d'oh

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@27.2.2-alpha.68a316.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://602ed0c7b96f443755735afd--gamut-preview.netlify.app>https://602ed0c7b96f443755735afd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/602ed0c7b96f443755735afd>Deploy Logs</a></p>

**Review by codecaaron — APPROVED** [review]
🚀

**Review by JoshuaKGoldberg — APPROVED** [review]
🚢

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@29.3.4-alpha.f31cff.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60351bdeaef4b52cf9e64da7--gamut-preview.netlify.app>https://60351bdeaef4b52cf9e64da7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60351bdeaef4b52cf9e64da7>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@29.3.5-alpha.f4c9ce.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60351d5bb8b2f12d2dd3c779--gamut-preview.netlify.app>https://60351d5bb8b2f12d2dd3c779--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60351d5bb8b2f12d2dd3c779>Deploy Logs</a></p>

**Pull request merged**